packages/react-charting/package.json

```diff
@@ -43,27 +43,27 @@
     "@fluentui/react-focus": "^8.8.44",
     "@fluentui/theme-samples": "^8.7.145",
     "@microsoft/load-themed-styles": "^1.10.26",
-    "@types/d3-array": "2.12.1",
-    "@types/d3-axis": "1.0.10",
-    "@types/d3-format": "^1.3.1",
-    "@types/d3-hierarchy": "2.0.0",
-    "@types/d3-sankey": "^0.11.0",
-    "@types/d3-scale": "^4.0.0",
-    "@types/d3-selection": "1.4.1",
-    "@types/d3-shape": "3.1.1",
-    "@types/d3-time-format": "^3.0.0",
-    "@types/d3-time": "^2.1.1",
+    "@types/d3-array": "2",
+    "@types/d3-axis": "2",
+    "@types/d3-format": "2",
+    "@types/d3-hierarchy": "2",
+    "@types/d3-sankey": "^0.12.3",
+    "@types/d3-scale": "4",
+    "@types/d3-selection": "2",
+    "@types/d3-shape": "3",
+    "@types/d3-time-format": "3",
+    "@types/d3-time": "2",
     "@fluentui/set-version": "^8.2.16",
-    "d3-array": "2.12.1",
-    "d3-axis": "1.0.8",
-    "d3-format": "^1.4.4",
-    "d3-hierarchy": "2.0.0",
+    "d3-array": "2",
```

Comment:
using fixed versions for dependencies will probably produce duplicates within applications. can we use either `^` or `~`. ( note: I'm not familiar how can we rely on d3 family of libraries following proper semver, that's why either ^ or ~ is the safe bet )

Reply:
Using ~ locks the major version and minor versions like `a.b.x` allowing patch `>= x`
Using ^ locks the major version and specifies the minimum supported version `a.x1.y2` with `x >= x1` and `y >= y1`.
For our requirements we want to set `x1 = 0` and `y2 = 0` meaning `>= a.0.0` which is same as just using `a` as we have used.

This paradigm is same as used by d3. 
See https://github.com/d3/d3/blob/v6.7.0/package.json with an exception of the d3-scale package.

d3 version convention is neatly defined.
Adopting their version convention standardizes our package dependency closer to d3 and reduces the chances of package duplications if an application uses other libraries depending on d3. This has been observed in teams and loop applications.
Even the d3-scale version 4.x.x allows older versions of d3-shape and d3-path which we will be able to benefit as part of this change.

Reply:
> Using ^ locks the major version and specifies the minimum supported version a.x1.y2 with x >= x1 and y >= y1.

`>= a.0.0` is not what you might thing it is, that would allow any major version starting with `a` which might be not compatible and cause issues.

Using just a major version is the same as `^Major.lowesMinor.lowestPatch` but it doesn't express the intent as clearly as using caret, thus I'd be great to follow our best practices with fluent packages and explicitly provide caret.

Reply:
pr approved. I'll leave this up to your decision/consideration. ty

Reply:
The behavior of using `a` and `^a.0.0` is the same and works without issues for our use case. You can see the changes to yarn.lock in the last commit.
Another reason is that we will soon provide a supported range of d3 dependencies to make our library more standard and align to co exist with other charting frameworks.

As fluent prefers to use ^, I have updated the version names to represent it. Thanks for the suggestions.